### PR TITLE
Feature/fallback return

### PR DESCRIPTION
**Pull request by novaknole**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: [ID]()

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/contracts/test/CallbackHandlerHelperMock.sol
Comment (brickpop) [inline packages/contracts/contracts/test/CallbackHandlerHelperMock.sol:15]: Does the stack-saving practice apply here?

```suggestion
    function handleCallback(bytes4 selector, bytes memory data) external returns (bytes4 magicNumber) {
        magicNumber = _handleCallback(selector, data);      
```

Comment (novaknole) [reply packages/contracts/contracts/test/CallbackHandlerHelperMock.sol:15]: It's a test file, so I preferred to be more explicit.

File: packages/contracts/contracts/test/CallbackHandlerHelperMock.sol
Comment (mathewmeconry) [inline packages/contracts/contracts/test/CallbackHandlerHelperMock.sol:18]:
```suggestion
    /// @notice Executes `_registerCallback` on the parent to register magic number per selector.
```

File: packages/contracts/contracts/test/CallbackHandlerHelperMock.sol
Comment (mathewmeconry) [inline packages/contracts/contracts/test/CallbackHandlerHelperMock.sol:19]:
```suggestion
    /// @param selector The function selector.
```

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

**Review by brickpop — APPROVED** [review]

File: packages/contracts/contracts/core/DAO.sol
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/core/DAO.sol:282]:
```suggestion
    /// @param _input An alias being equivalent to `msg.data`. This a feature of the fallback function was introduced with [solidity compiler version 0.7.6](https://github.com/ethereum/solidity/releases/tag/v0.7.6)
    /// @return bytes The magic number registered for the function selector triggering the fallback.
```

File: packages/contracts/contracts/core/component/CallbackHandler.sol
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/core/component/CallbackHandler.sol:29]:
```suggestion
    /// @param _callbackSelector The function selector of the callback function.
    /// @return The magic number registered for the function selector triggering the fallback.
```

File: packages/contracts/contracts/core/component/CallbackHandler.sol
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/core/component/CallbackHandler.sol:39]:
```suggestion
        emit CallbackReceived({sig: _callbackSelector, data: _data});
```

File: packages/contracts/contracts/test/CallbackHandlerHelperMock.sol
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/test/CallbackHandlerHelperMock.sol:12]:
```suggestion
    /// @notice Calls the internal `_handleCallback` on the parent `CallbackHandler` for testing purposes.
    /// @param selector The function selector of the callback function to be tested.
    /// @param data Arbitrary data accompanying the callback that will be emitted with the `CallbackReceived` event.
```

**Review by Michael-A-Heuer — CHANGES_REQUESTED** [review]
Looking good. Can be approved once NatSpec suggestions are merged.

File: packages/contracts/contracts/core/DAO.sol
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/core/DAO.sol:281]:
```suggestion
    /// @param _input An alias being equivalent to `msg.data`. This feature of the fallback function was introduced with [solidity compiler version 0.7.6](https://github.com/ethereum/solidity/releases/tag/v0.7.6)
```

File: packages/contracts/contracts/core/component/CallbackHandler.sol
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/core/component/CallbackHandler.sol:29]:
```suggestion
    /// @return The magic number registered for the function selector triggering the callback.
```

**Review by Michael-A-Heuer — APPROVED** [review]

**Review by brickpop — APPROVED** [review]

**Pull request merged**
